packages/wallet/api/src/lib-wallet.js

```diff
@@ -556,11 +557,42 @@ export function makeWallet({
       });
   }
 
+  /**
+   * @param { () => ( Promise | undefined ) } get - The function whose return value
+   * to memoize.
+   */
+  const makeMemoizedGetter = get => {
+    let pk;
+    const doGet = () => pk.resolve(get());
+
+    return () => {
+      if (pk === undefined) {
+        pk = makePromiseKit();
+        doGet();
+      }
+      return pk.promise;
+    };
+  };
+
+  const getAttIssuer = makeMemoizedGetter(() =>
+    E(agoricNames)?.lookup('issuer', 'Attestation'),
+  );
+
+  const getAttBrand = makeMemoizedGetter(() => E(getAttIssuer()).getBrand());
```

Comment:
This doesn't need a function def here does it? Can't we pass `getBrand` itself?

Reply:
Why wouldn't we want to get them at wallet initialization time? Will they change? Is it a perf optimization? Does wallet init block some UI?

Reply:
They shouldn't change, it's to keep the wallet initialization slim. Maybe it's an unnecessary optimization, but wallet init blocks `agoric start --reset` and eventually whatever UI we have for deploying a new wallet on-chain.